File: handlerclaw/lib/core/data/mappers/user_mapper.dart
import 'package:handlerclaw/core/domain/entities/user_entity.dart';
import 'package:handlerclaw/core/data/dto/user_dto.dart';

class UserMapper {
  static UserEntity fromDto(UserDto dto) {
    return UserEntity(
      id: dto.id,
      name: dto.name,
      email: dto.email,
      lastLoginAt: dto.lastLoginAt,
      createdAt: dto.createdAt,
      updatedAt: dto.updatedAt,
    );
  }

  static UserDto toDto(UserEntity entity) {
    return UserDto(
      id: entity.id,
      name: entity.name,
      email: entity.email,
      lastLoginAt: entity.lastLoginAt,
      createdAt: entity.createdAt,
      updatedAt: entity.updatedAt,
    );
  }
}

File: api/src/modules/auth/infrastructure/mappers/user.mapper.ts
import { type User as PrismaUser } from "../../../../lib/generated/prisma/client.js";
import type { IUser } from "../../domain/entities/user.entity.js";

export function toUserEntity(data: PrismaUser): IUser {
  return {
    id: data.id,
    email: data.email,
    name: data.name,
    last_login_at: data.lastLoginAt,
    created_at: data.createdAt,
    updated_at: data.updatedAt,
  };
}
